### PR TITLE
feat: update samples to use @okta/okta-react-native v1.4.0

### PR DESCRIPTION
**Pull request by shuowu-okta**
What's in this PR:

- Update both samples to use @okta/okta-react-native v1.4.0
- Remove @okta/okta-auth-js dependency for custom-sign-in sample
- Remove manual link for @okta/okta-react-native, because react-native started support autolink since v0.60
- Update react-navigation module to v5
- Update READMEs accordingly

### CODE REVIEW

**Comment by IldarAbdullin-okta** [comment]
I don't see too much benefits of adding additional cross language communication. Auth-js perfectly handles AuthN cases and you don't need to leave javascript context to handle AuthN states. So your whole implementation can be js without extra interaction with native layer.

@chrismorris-okta , @robertdamphousse-okta , thoughts?

File: custom-sign-in/package.json [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
+    "android": "react-native run-android",
+    "ios": "react-native run-ios",
     "test": "jest",
     "lint": "eslint ."
   },
   "dependencies": {
     "@okta/configuration-validation": "^0.4.1",
-    "@okta/okta-auth-js": "2.11.2",
-    "@okta/okta-react-native": "^1.2.1",
+    "@okta/okta-react-native": "file:okta-okta-react-native-v1.4.0.tgz",
```
Comment (shuowu-okta) [inline custom-sign-in/package.json:14]: Need update dependency when new release available.

File: README.md [hunk 2]
```diff
@@ -4,6 +4,47 @@ This repository contains several sample applications that demonstrate various Ok
...
+## Prerequisites
+
+Before running the samples, you will need the following:
+
+* An Okta Developer Account, you can sign up for one at <https://developer.okta.com/signup/>.
+* An Okta Application, configured for Native mode. This is done from the Okta Developer Console. After login, from the Admin dashboard, navigate to `Applications → Add Application`. Choose Native as the platform. Populate your new Native OpenID Connect application with values similar to:
+  * **Application Name** 
+    * Native OpenId Connect App (must be unique)
+  * **Login redirect URIs**
+    * `com.sampleapplication:/`
```
Comment (swiftone) [inline README.md:16]: I know this is consistent with our other mobile guides, but is there any better guidance for them on what to do in an actual production app?  Will this be understood by most developers, or will they make a lot of "com.sampleapplication" entries?

File: README.md [hunk 3]
```diff
@@ -4,6 +4,47 @@ This repository contains several sample applications that demonstrate various Ok
...
+* **Issuer/Discovery Uri** - This is the URL of the authorization server that will perform authentication.  All Developer Accounts have a "default" authorization server.  The issuer is a combination of your Org URL (found in the upper right of the console home page) and `/oauth2/default`. For example, `https://dev-1234.oktapreview.com/oauth2/default`.
+
+Now place these values into the file `samples.config.js` located under each sample folder:
+
+```javascript
+export default {
+  oidc: {
+    clientId: '{clientId}',
+    discoveryUri: 'https://{yourOktaDomain}.com/oauth2/default',
+    redirectUri: 'com.sampleapplication:/',
```
Comment (swiftone) [inline README.md:38]: We should likewise consider updating here.

**Comment by shuowu-okta** [comment]
> I don't see too much benefits of adding additional cross language communication. Auth-js perfectly handles AuthN cases and you don't need to leave javascript context to handle AuthN states. So your whole implementation can be js without extra interaction with native layer.
> 
> @chrismorris-okta , @robertdamphousse-okta , thoughts?

Decided to move to use authJS for AuthN communication. Both @okta/okta-auth-js and @okta/okta-react-native had been updated and published.

File: README.md (hunk 2 above)
Comment (shuowu-okta) [reply README.md:16]: Make sense. Here I just moved it up into the root level, how about wrap it with bracket, so devs may pay attention.

**Comment by aarongranick-okta** [comment]
I am seeing both yarn.lock and package-lock.json files. We should only have one of these.

**Comment by shuowu-okta** [comment]
> I am seeing both yarn.lock and package-lock.json files. We should only have one of these.

yarn.lock in diff is the deleted file.

**Pull request merged**